builtin/providers/aws/resource_aws_lambda_function.go

```diff
@@ -323,15 +323,26 @@ func resourceAwsLambdaFunctionRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("runtime", function.Runtime)
 	d.Set("timeout", function.Timeout)
 	d.Set("kms_key_arn", function.KMSKeyArn)
-	if config := flattenLambdaVpcConfigResponse(function.VpcConfig); len(config) > 0 {
-		log.Printf("[INFO] Setting Lambda %s VPC config %#v from API", d.Id(), config)
-		err := d.Set("vpc_config", config)
-		if err != nil {
-			return fmt.Errorf("Failed setting vpc_config: %s", err)
-		}
+
+	config := flattenLambdaVpcConfigResponse(function.VpcConfig)
+	log.Printf("[INFO] Setting Lambda %s VPC config %#v from API", d.Id(), config)
+	vpcSetErr := d.Set("vpc_config", config)
+	if vpcSetErr != nil {
+		return fmt.Errorf("Failed setting vpc_config: %s", vpcSetErr)
```

Comment:
Example DEBUG output of said empty vpc:

```
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws: Function:
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws: {
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   CodeSha256: "Ux/n9CP8l+7Ht0tICw0QPs0yLdC1b+1nJ9K5MZR9ENw=",
					[ ... ]
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   Role: "arn:aws:iam::187416307283:role/cts-iam-lambda",
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   Runtime: "nodejs",
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   Timeout: 3,
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   Version: "$LATEST",
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   VpcConfig: {
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:     SecurityGroupIds: [],
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:     SubnetIds: [],
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:     VpcId: ""
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws:   }
2016/11/21 17:03:41 [DEBUG] plugin: terraform-provider-aws: }
```

Only happens if I edit the ENV vars in the Web ui (in this case I added one)